universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java

```diff
@@ -198,6 +227,52 @@ private ContainerTag renderBlockList(
     return container;
   }
 
+  private ContainerTag blockMoveButtons(
+      Request request,
+      long programId,
+      ImmutableList<BlockDefinition> blockDefinitions,
+      BlockDefinition blockDefinition) {
+    String moveUpFormAction =
+        routes.AdminProgramBlocksController.move(programId, blockDefinition.id()).url();
+    // Move up button is invisible for the first block
+    String moveUpStyles =
+        StyleUtils.joinStyles(
+            AdminStyles.MOVE_BLOCK_BUTTON,
+            (blockDefinition.id() == blockDefinitions.get(0).id()) ? Styles.INVISIBLE : "");
+    Tag moveUp =
+        div()
+            .with(
+                form()
+                    .withAction(moveUpFormAction)
+                    .withMethod(HttpVerbs.POST)
+                    .with(makeCsrfTokenInputTag(request))
+                    .with(input().isHidden().withName("direction").withValue(Direction.UP.name()))
+                    .with(submitButton("^").withClasses(moveUpStyles)));
+
+    String moveDownFormAction =
+        routes.AdminProgramBlocksController.move(programId, blockDefinition.id()).url();
+    // Move down button is invisible for the last block
+    String moveDownStyles =
+        StyleUtils.joinStyles(
+            AdminStyles.MOVE_BLOCK_BUTTON,
+            (blockDefinition.id() == blockDefinitions.get(blockDefinitions.size() - 1).id())
```

Comment:
Same as above, but `last:`: https://tailwindcss.com/docs/hover-focus-and-other-states#last-child

Reply:
Can't really use tailwind's `last` because of how deeply nested the element we want to make invisible is.